radsat-sk/operation/subsystems/RBattery.c

```
@@ -0,0 +1,221 @@
+/***************************************************************************************************
+                                          PRIVATE GLOBALS
+***************************************************************************************************/
+
+/**
+ * 3 Byte commands for requesting output voltage readings for each bus
+ */
+static uint32_t batteryVoltageCommandBytes[3] = {
+		0x10E280,		// Output Voltage of Battery
+		0x10E210,		// Output Voltage of 5V Bus
+		0x10E200		// Output Voltage of 3.3V Bus
+};
+
+/**
+ * 3 Byte commands for requesting output current readings for each bus
+ */
+static uint32_t batteryCurrentCommandBytes[3] = {
+		0x10E284,		// Output Current of Battery Bus in mA
+		0x10E214,		// Output Current of 5V Bus in mA
+		0x10E204		// Output Current of 3.3V Bus in mA
+};
+
+/**
+ * 4 Byte commands for requesting temperature readings for each board
+ */
+static uint32_t batteryTemperatureCommandBytes[4] = {
+		0x10E308,		// Motherboard Temperature
+		0x10E398,		// Daughterboard 1 Temperature
+		0x10E3A8,		// Daughterboard 2 Temperature
+		0x10E3B8		// Daughterboard 3 Temperature
+};
+
+/***************************************************************************************************
+                                       PRIVATE FUNCTION STUBS
+***************************************************************************************************/
+
+static uint16_t batteryI2cTalk(uint32_t command);
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Request and store all of the relavant data (voltage, current, temperature) from the battery board
+ * 		and battery daughter boards
+ * @return A BatteryStatus object containing all of the telemetry of the battery board
+ */
+BatteryStatus getBatteryTelemetry(void){
```

Same idea was mentioned in the PDB PR, you want to return the error and pass a pointer to a struct as a param instead